src/App.tsx

```diff
@@ -15,37 +17,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+export const BY_ALPHABETICALLY = 'alphabetically';
+export const BY_LENGTH = 'length';
+export const BY_REVERSE = 'reverse';
+
+type SortField = typeof BY_ALPHABETICALLY | typeof BY_LENGTH | null;
+
+export function getSortList(
+  goods: string[],
+  sortField: SortField,
+  reverse: boolean,
+): string[] {
+  const sortGoods = [...goods];
+
+  switch (sortField) {
+    case BY_ALPHABETICALLY:
+      sortGoods.sort();
+      break;
+    case BY_LENGTH:
+      sortGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+    default:
+      break;
+  }
+
+  if (reverse) {
+    sortGoods.reverse();
+  }
+
+  return sortGoods;
+}
```

Comment:
Although the sorting function is implemented correctly, the checklist emphasizes not to interact with the DOM directly and to use React as much as possible. Since this function is pure and doesn't interact with the DOM, it's fine, but make sure that all DOM interactions happen through React state and props, not directly.